src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java

```diff
@@ -402,25 +402,30 @@ private void validateCmekKey(FailureCollector failureCollector) {
       try {
         CryptoKeyName cmekKeyName = CryptoKeyName.parse(cmekKey);
         Boolean isServiceAccountFilePath = isServiceAccountFilePath();
-        if (isServiceAccountFilePath != null) {
-          Credentials credentials = getServiceAccount() == null ?
-            null : GCPUtils.loadServiceAccountCredentials(getServiceAccount(), isServiceAccountFilePath);
-          Storage storage = GCPUtils.getStorage(getProject(), credentials);
-          Bucket bucket = null;
-          try {
-            bucket = storage.get(getBucket());
-          } catch (StorageException e) {
-            return;
+        Credentials credentials = getServiceAccount() == null ?
+          null : GCPUtils.loadServiceAccountCredentials(getServiceAccount(), isServiceAccountFilePath);
+        Storage storage = GCPUtils.getStorage(getProject(), credentials);
+        Bucket bucket = null;
+        try {
+          bucket = storage.get(getBucket());
+        } catch (StorageException e) {
+          /* Ignoring the exception because we don't want the validation to fail if there is an exception getting
+            the bucket information either because the service account used during validation is the CDF service account
```

Comment:
In general, don't mention CDF or make assumptions about where the code is running. This is the open source code that can run in all sorts of environments. Change to:

... during validation can be different than the service account that will be used at runtime.